x-pack/plugin/esql/compute/gen/src/main/java/org/elasticsearch/compute/gen/AggregatorImplementer.java

```diff
@@ -119,7 +121,8 @@ public AggregatorImplementer(
             return a;
         }).filter(a -> a instanceof PositionArgument == false).toList();
 
-        this.hasOnlyBlockArguments = this.aggParams.stream().allMatch(a -> a instanceof BlockArgument);
+        this.tryToUseVectors = aggParams.stream().anyMatch(a -> (a instanceof BlockArgument) == false)
+            && aggParams.stream().noneMatch(a -> a instanceof StandardArgument && a.dataType(false) == null);
```

Comment:
I'd prefer making a method in `Argument` that's, like, `hasVector` or something.

Reply:
Fixed in https://github.com/elastic/elasticsearch/pull/137459/commits/00d6d781978340ce7f1f416a4935e7777e187c38.